projects/plugins/jetpack/modules/sso.php

```diff
@@ -83,6 +88,52 @@ private function __construct() {
 		}
 	}
 
+	/**
+	 * Render the invitation email message.
+	 */
+	public function render_invitation_email_message() {
+		printf( '<div class="notice inline notice-large notice-info">%1s</div>', esc_html__( 'The user will be automatically invited to WordPress.com.', 'jetpack' ) );
+	}
+
+	/**
+	 * Intercept the core invitation email.
+	 *
+	 * @param bool    $notify Whether to notify the user or not.
+	 * @param WP_User $user   The user object.
+	 *
+	 * @return bool Whether to notify the user or not via core email.
+	 */
+	public function intercept_core_invitation_email( $notify, $user ) {
+		if ( ! $user instanceof WP_User ) {
+			return $user;
+		}
+
+		$email   = $user->user_email;
+		$role    = $user->roles[0];
+		$locale  = get_user_locale( $user->ID );
+		$blog_id = Jetpack_Options::get_option( 'id' );
+		$url     = '/sites/' . $blog_id . '/invites/new';
+		$url     = add_query_arg( 'locale', $locale, $url );
+
+		Client::wpcom_json_api_request_as_user(
+			$url,
+			'2', // Api version
+			array(
+				'method' => 'POST',
+			),
+			array(
+				'invitees' => array(
+					array(
+						'email_or_username' => $email,
+						'role'              => $role,
+					),
+				),
+			)
+		);
+		// returning false prevents the user to be notified by the core email.
+		return false;
```

Comment:
We should investigate some error handling. We may want to block adding a user unless the invite works because otherwise the user won't be able to log in any way.

Reply:
This might be tricky, if the invite fails or the user does not accept or reply to the email invitation is mostly the same. I guess in this case, the user status would be pending.